mantidimaging/gui/windows/operations/presenter.py

```diff
@@ -196,8 +193,25 @@ def _post_filter(self, updated_stacks, task):
             self.view.clear_notification_dialog()
             self.view.show_operation_completed(self.model.selected_filter.filter_name)
 
-    def _do_apply_filter(self, apply_to):
-        self.model.do_apply_filter(apply_to, partial(self._post_filter, apply_to))
+    def _do_apply_filter(self, apply_to, allow_180_degree=False, confirm_with_user_for_180degree=False):
+        confirmed_stacks = []
+        for stack in apply_to:
+            is_180_proj = self.is_a_proj180deg(stack)
+            if not is_180_proj:
+                confirmed_stacks.append(stack)
+            elif allow_180_degree:
+                confirmed_stacks.append(stack)
+            elif confirm_with_user_for_180degree and\
+                self.view.ask_confirmation("Operations applied to the sample are also automatically applied to the 180 "
+                                           "degree projection. Please avoid applying an operation unless you're "
+                                           "absolutely certain you need to.\nAre you sure you want to apply to 180 "
+                                           "degree projection?"):
+                confirmed_stacks.append(stack)
```

Comment:
I'm still not convinced we need this here when the only thing that uses it is `do_apply_filter`. 

All other paths ignore the choices via the function parameters:
- Do apply on 180 proj comes with allow 180 and no confirm
- DO apply all comes with don't allow and don't confirm

So it seems this is better placed inside `do_apply_filter` itself, rather than cluttering this function